lib/resty/auto-ssl/init_master.lua

```diff
@@ -28,9 +28,16 @@ end
 -- secret token is an extra precaution to ensure the server is not accidentally
 -- opened up or proxied to the outside world.
 local function generate_hook_sever_secret()
+  if ngx.shared.auto_ssl_settings:get("hook_server:secret") then
+    -- if we've already got a secret token, do not overwrite it, as this causes
+    -- problems in reload-heavy envrionments.
+    -- See https://github.com/GUI/lua-resty-auto-ssl/issues/66
+    return
+  end
+
   -- Generate the secret token.
   local random = resty_random.bytes(32)
-  AUTO_SSL_HOOK_SECRET = str.to_hex(random)
+  ngx.shared.auto_ssl_settings:set("hook_server:secret", str.to_hex(random))
```

Comment:
After the issues we saw earlier with not a lot of error logging when shard dicts run out of memory, I've just been trying to wrap any `:set` with additional error logging, like in https://github.com/GUI/lua-resty-auto-ssl/commit/3eb12de6d20feaabf81eba24453a5e5c5177b0ec

So maybe something like:

```lua
local _, set_err, set_forcible = ngx.shared.auto_ssl_settings:set("hook_server:secret", str.to_hex(random))
if set_err then
  ngx.log(ngx.ERR, "auto-ssl: failed to set shdict for hook_server:secret: ", set_err)
elseif set_forcible then
  ngx.log(ngx.ERR, "auto-ssl: 'lua_shared_dict auto_ssl_settings' might be too small - consider increasing its configured size (old entries were removed while adding hook_server:secret)")
end
```